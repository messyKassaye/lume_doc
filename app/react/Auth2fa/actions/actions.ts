/** @format */
import { Dispatch } from 'redux';

import { User } from 'api/users/usersModel';
import { actions } from 'app/BasicReducer';
import { notificationActions } from 'app/Notifications';
import { RequestParams } from 'app/utils/RequestParams';

import Auth2faAPI from '../Auth2faAPI';

export interface enable2faType {
  type: string;
  key: any;
  value: any;
}

export function enable2fa(): enable2faType {
  return actions.setIn('auth/user', 'using2fa', true);
}

export function reset2fa(user: User) {
  return async (dispatch: Dispatch<{}>) => {
    await Auth2faAPI.reset2fa(new RequestParams({ _id: user._id }));
    dispatch(actions.update('users', { ...user, using2fa: false }));
    dispatch(notificationActions.notify('Deleted successfully.', 'success'));
  };
}
