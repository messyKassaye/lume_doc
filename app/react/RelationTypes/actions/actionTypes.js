//DATA
export const RELATION_TYPE_SAVED = 'RELATION_TYPE_SAVED';
export const EDIT_RELATION_TYPE = 'EDIT_RELATION_TYPE';
export const RELATION_TYPE_DELETED = 'RELATION_TYPE_DELETED';
export const SET_RELATION_TYPES = 'SET_RELATION_TYPES';
export const RESET_RELATION_TYPE = 'RESET_RELATION_TYPE';
