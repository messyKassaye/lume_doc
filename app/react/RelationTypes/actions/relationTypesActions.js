import api from 'app/RelationTypes/RelationTypesAPI';
import { actions as formActions } from 'react-redux-form';
import referencesAPI from 'app/Viewer/referencesAPI';
import { actions } from 'app/BasicReducer';
import { RequestParams } from 'app/utils/RequestParams';

export function editRelationType(relationType) {
  return formActions.load('template.data', relationType);
}

export function deleteRelationType(relationType) {
  return function(dispatch) {
    return api.delete(new RequestParams({ _id: relationType._id })).then(() => {
      dispatch(actions.remove('relationTypes', relationType));
    });
  };
}

export function checkRelationTypeCanBeDeleted(relationType) {
  return function() {
    return referencesAPI
      .countByRelationType(new RequestParams({ relationtypeId: relationType._id }))
      .then(count => {
        if (count) {
          return Promise.reject();
        }
      });
  };
}
