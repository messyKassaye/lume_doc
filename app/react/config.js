const APIURL = process.env.API_URL || '/api/';
export default { APIURL };
export { APIURL };
