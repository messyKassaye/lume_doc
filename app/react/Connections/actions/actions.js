import { actions } from 'app/BasicReducer';
import { notificationActions } from 'app/Notifications';
import api from 'app/utils/api';
import debounce from 'app/utils/debounce';
import { RequestParams } from 'app/utils/RequestParams';

import * as types from './actionTypes';
import * as uiActions from './uiActions';

export function immidiateSearch(dispatch, searchTerm, connectionType) {
  dispatch(uiActions.searching());

  const requestParams = new RequestParams({ searchTerm, fields: ['title'] });

  return api.get('search', requestParams).then(response => {
    let results = response.json.rows;
    if (connectionType === 'targetRanged') {
      results = results.filter(r => r.documents.length);
    }
    dispatch(actions.set('connections/searchResults', results));
  });
}

const debouncedSearch = debounce(immidiateSearch, 400);

export function search(searchTerm, connectionType) {
  return dispatch => {
    dispatch(actions.set('connections/searchTerm', searchTerm));
    return debouncedSearch(dispatch, searchTerm, connectionType);
  };
}

export function startNewConnection(connectionType, sourceDocument) {
  return dispatch =>
    immidiateSearch(dispatch, '', connectionType).then(() => {
      dispatch(actions.set('connections/searchTerm', ''));
      dispatch(uiActions.openPanel(connectionType, sourceDocument));
    });
}

export function setRelationType(template) {
  return {
    type: types.SET_RELATION_TYPE,
    template,
  };
}

export function setTargetDocument(id) {
  return {
    type: types.SET_TARGET_DOCUMENT,
    id,
  };
}

export function saveConnection(connection, callback = () => {}) {
  return (dispatch, getState) => {
    dispatch({ type: types.CREATING_CONNECTION });
    if (connection.type !== 'basic') {
      connection.language = getState().locale;
    }

    delete connection.type;

    const sourceRelationship = {
      entity: connection.sourceDocument,
      template: null,
      reference: connection.sourceRange,
      file: connection.sourceFile,
    };

    const targetRelationship = { entity: connection.targetDocument, template: connection.template };
    if (connection.targetRange) {
      Object.assign(targetRelationship, {
        reference: connection.targetRange,
        file: connection.targetFile,
      });
    }

    const apiCall = {
      delete: [],
      save: [[sourceRelationship, targetRelationship]],
    };

    return api.post('relationships/bulk', new RequestParams(apiCall)).then(response => {
      dispatch({ type: types.CONNECTION_CREATED });
      callback(response.json);
      dispatch(notificationActions.notify('saved successfully !', 'success'));
    });
  };
}

export function selectRangedTarget(connection, onRangedConnect) {
  return dispatch => {
    dispatch({ type: types.CREATING_RANGED_CONNECTION });
    onRangedConnect(connection.targetDocument);
  };
}
