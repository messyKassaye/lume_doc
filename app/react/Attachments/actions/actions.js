import { actions as formActions } from 'react-redux-form';
import superagent from 'superagent';

import { APIURL } from 'app/config.js';
import { notify } from 'app/Notifications/actions/notificationsActions';
import { updateEntity, selectSingleDocument } from 'app/Library/actions/libraryActions';
import api from 'app/utils/api';
import { RequestParams } from 'app/utils/RequestParams';
import { actions as basicReducerActions } from 'app/BasicReducer';

import * as types from './actionTypes';

export function updateFile(file, entity) {
  return async dispatch => {
    await api.post('files', new RequestParams(file));
    const documents = entity.documents.map(f => {
      if (f._id === file._id) {
        return file;
      }
      return f;
    });
    const updatedEntity = Object.assign(entity, { documents });
    dispatch(basicReducerActions.set('viewer/doc', updatedEntity));
    dispatch(updateEntity(updatedEntity));
    await dispatch(selectSingleDocument(updatedEntity));
    dispatch(notify('File updated', 'success'));
  };
}

export function deleteFile(file, entity) {
  return async dispatch => {
    await api.delete('files', new RequestParams({ _id: file._id }));
    const documents = entity.documents.filter(f => f._id !== file._id);

    const updatedEntity = Object.assign(entity, { documents });
    dispatch(basicReducerActions.set('viewer/doc', updatedEntity));
    dispatch(updateEntity(updatedEntity));
    await dispatch(selectSingleDocument(updatedEntity));
    dispatch(notify('File deleted', 'success'));
  };
}

export function uploadAttachment(entity, file, __reducerKey) {
  return dispatch => {
    dispatch({ type: types.START_UPLOAD_ATTACHMENT, entity });
    superagent
      .post(`${APIURL}files/upload/attachment`)
      .set('Accept', 'application/json')
      .set('X-Requested-With', 'XMLHttpRequest')
      .field('entity', entity)
      .attach('file', file, file.name)
      .on('progress', data => {
        dispatch({ type: types.ATTACHMENT_PROGRESS, entity, progress: Math.floor(data.percent) });
      })
      .on('response', result => {
        dispatch({
          type: types.ATTACHMENT_COMPLETE,
          entity,
          file: JSON.parse(result.text),
          __reducerKey,
        });
      })
      .end();
  };
}

export function uploadAttachmentFromUrl(entity, name, url, __reducerKey) {
  return dispatch => {
    dispatch({ type: types.START_UPLOAD_ATTACHMENT, entity });
    api
      .post('files', new RequestParams({ originalname: name, url, entity, type: 'attachment' }))
      .then(newFile => {
        dispatch({ type: types.ATTACHMENT_COMPLETE, entity, file: newFile.json, __reducerKey });
        dispatch(notify('Attachment uploaded', 'success'));
      });
  };
}

export function renameAttachment(entityId, form, __reducerKey, file) {
  return dispatch =>
    api
      .post(
        'files',
        new RequestParams({
          _id: file._id,
          originalname: file.originalname,
        })
      )
      .then(renamedFile => {
        dispatch({
          type: types.ATTACHMENT_RENAMED,
          entity: entityId,
          file: renamedFile.json,
          __reducerKey,
        });
        dispatch(formActions.reset(form));
        dispatch(notify('Attachment renamed', 'success'));
      });
}

export function deleteAttachment(entitySharedId, attachment, __reducerKey) {
  return async dispatch => {
    await api.delete(
      'files',
      new RequestParams({
        _id: attachment._id,
      })
    );
    dispatch({
      type: types.ATTACHMENT_DELETED,
      entity: entitySharedId,
      file: attachment,
      __reducerKey,
    });

    dispatch(notify('Attachment deleted', 'success'));
  };
}

export function loadForm(form, attachment) {
  return dispatch => {
    dispatch(formActions.reset(form));
    dispatch(formActions.load(form, attachment));
  };
}

export function submitForm(form) {
  return dispatch => {
    dispatch(formActions.submit(form));
  };
}

export function resetForm(form) {
  return dispatch => {
    dispatch(formActions.reset(form));
  };
}
