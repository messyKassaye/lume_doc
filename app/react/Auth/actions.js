/** @format */

import api from 'app/utils/api';
import { actions } from 'app/BasicReducer';
import { notify } from 'app/Notifications/actions/notificationsActions';
import { RequestParams } from 'app/utils/RequestParams';

export function login(credentials) {
  const request = new RequestParams(credentials);
  return async dispatch => {
    await api.post('login', request);
    const user = await api.get('user');
    dispatch(actions.set('auth/user', user.json));
  };
}

export function recoverPassword(email) {
  const request = new RequestParams({ email });
  return dispatch =>
    api.post('recoverpassword', request).then(() => {
      dispatch(
        notify(
          'Instructions to reset your password have been send, please check your email',
          'success'
        )
      );
    });
}

export function resetPassword(password, key) {
  const request = new RequestParams({ password, key });
  return dispatch =>
    api.post('resetpassword', request).then(() => {
      dispatch(notify('Password changed success', 'success'));
    });
}

export function unlockAccount(credentials) {
  const request = new RequestParams(credentials);
  return dispatch =>
    api.post('unlockaccount', request).then(() => {
      dispatch(notify('Account unlocked successfully', 'success'));
    });
}
