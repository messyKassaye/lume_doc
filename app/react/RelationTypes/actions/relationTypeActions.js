import * as types from 'app/RelationTypes/actions/actionTypes';
import api from 'app/RelationTypes/RelationTypesAPI';
import { notificationActions } from 'app/Notifications';
import { t } from 'app/I18N';
import { RequestParams } from 'app/utils/RequestParams';

export function saveRelationType(relationType) {
  return dispatch =>
    api.save(new RequestParams(relationType)).then(() => {
      dispatch({ type: types.RELATION_TYPE_SAVED });
      dispatch(
        notificationActions.notify(t('System', 'RelationType saved', null, false), 'success')
      );
    });
}

export function resetRelationType() {
  return { type: types.RESET_RELATION_TYPE };
}
