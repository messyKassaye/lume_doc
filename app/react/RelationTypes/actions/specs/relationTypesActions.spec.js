/* eslint-disable max-nested-callbacks */
import backend from 'fetch-mock';
import { APIURL } from 'app/config.js';

import referencesAPI from 'app/Viewer/referencesAPI';
import * as modalActions from 'app/Modals/actions/modalActions';
import * as actions from 'app/RelationTypes/actions/relationTypesActions';

describe('relationTypesActions', () => {
  describe('editRelationType', () => {
    it('should load the relationType in the form model', () => {
      const relationType = { name: 'Secret list of things', values: [] };
      expect(actions.editRelationType(relationType)).toMatchSnapshot();
    });
  });

  describe('async action', () => {
    let dispatch;

    beforeEach(() => {
      backend.restore();
      backend.delete(`${APIURL}relationtypes?_id=relationTypeId`, {
        body: JSON.stringify({ testBackendResult: 'ok' }),
      });
      dispatch = jasmine.createSpy('dispatch');
    });

    afterEach(() => backend.restore());

    describe('deleteRelationType', () => {
      it('should delete the relationType and dispatch a relationTypes/REMOVE action with the relation type', done => {
        const relationType = { _id: 'relationTypeId' };
        actions
          .deleteRelationType(relationType)(dispatch)
          .then(() => {
            expect(dispatch).toHaveBeenCalledWith({
              type: 'relationTypes/REMOVE',
              value: relationType,
            });
            done();
          });
      });
    });

    describe('checkRelationTypeCanBeDeleted', () => {
      let relationType;
      beforeEach(() => {
        spyOn(modalActions, 'showModal');
        relationType = { _id: 'abc1' };
      });

      describe('when there is references using the relation type', () => {
        it('should showModal CantDeleteRelationType', done => {
          spyOn(referencesAPI, 'countByRelationType').and.returnValue(Promise.resolve(2));

          actions
            .checkRelationTypeCanBeDeleted(relationType)(dispatch)
            .then(() => {
              expect('this promise to not be rejected').toBe(false);
              done();
            })
            .catch(() => {
              done();
            });
        });
      });

      describe('when there is not references using the relation type', () => {
        it('should return a rejected promise', done => {
          spyOn(referencesAPI, 'countByRelationType').and.returnValue(Promise.resolve(0));

          actions
            .checkRelationTypeCanBeDeleted(relationType)(dispatch)
            .then(() => {
              done();
            })
            .catch(() => {
              expect('this promise to be rejected').toBe(false);
              done();
            });
        });
      });
    });
  });
});
