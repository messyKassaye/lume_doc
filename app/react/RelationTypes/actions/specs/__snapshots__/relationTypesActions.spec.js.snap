// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`relationTypesActions editRelationType should load the relationType in the form model 1`] = `
Object {
  "external": true,
  "load": true,
  "model": "template.data",
  "multi": false,
  "silent": true,
  "type": "rrf/change",
  "value": Object {
    "name": "Secret list of things",
    "values": Array [],
  },
}
`;
